cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataSubscriberService.java

```diff
@@ -66,7 +71,9 @@
 public class MetadataSubscriberService extends AbstractMessagingSubscriberService<MetadataMessage> {
```

Comment:
Currently we already have a subscriber service for metadata, and it can be used for the profile metadata. But it has a fetch size of default 100, and these messages are processed in one transaction. For our case, it will take some time if it is instance level metadata change, so it might throw a tx timeout. @albertshau @chtyim  do you think it is good to have a separate subsriber service for our case or we will simply increase the timeout of Transactional in this service?

Reply:
You can change AbstractMessagingSubscriberService.processMessages to return a message id instead of being null. It can return the last processed message in the iterator. That way, if you see a namespace or instance level profile change, you can make sure that's the only message you process in a single transaction.

This class seems to deal with a wide range of messages. Usage and lineage are related to each other, but not to workflow state change and workflow token saving. I guess it's ok to also handle profile messages here, not too sure what the trade-offs are between keeping it here or having a separate topic.

Reply:
so the iterator itself already contains the last message id, so i changed the MetadataSubscriberService to return if we encounter a message which is about namespace and instance level change